task/net/src/server_vlan.rs

```diff
@@ -202,10 +208,23 @@ impl<'a> ServerImpl<'a> {
             }
             *ifaces_iter.next().unwrap() = Some(iface);
 
-            // Increment the MAC and IP addresses so that each VLAN has
-            // a unique address.
-            ipv6_addr.0[15] += 1;
-            mac.0[5] += 1;
+            // Increment the MAC and IP addresses based on the stride in the
+            // configuration block, so that each VLAN has a unique address.
+            //
+            // We take advantage of the fact that the top 3 octets are the OUI,
+            // and convert into a u32 to do the incrementing.
+            let next_mac = (u32::from_be_bytes(mac[2..].try_into().unwrap())
+                & 0xFFFFFF)
+                + mac_address_block.stride as u32;
```

Comment:
The comment and code disagree here. You're including a byte from the OUI. If the API requires four bytes and that's a constraint, then make that clear in the comment as otherwise it's confusing and I'd ask why is it being included and masked off.

Reply:
Ah, it's a little funky: we need a `[u8; 4]` to call `u32::from_be_bytes`, so we include one byte of the OUI then mask it out.  I added a comment to that effect, because this confused me as well upon revisiting the code!

Reply:
Thanks, that looks good.